services/graph/pkg/service/v0/driveitems.go

```diff
@@ -124,13 +124,15 @@ func (g Graph) getRemoteItem(ctx context.Context, root *storageprovider.Resource
 		return nil, err
 	}
 
-	if baseURL != nil {
+	if baseURL != nil && res.GetInfo() != nil && res.GetInfo().GetSpace() != nil {
 		// TODO read from StorageSpace ... needs Opaque for now
 		// TODO how do we build the url?
 		// for now: read from request
 		webDavURL := *baseURL
-		webDavURL.Path = path.Join(webDavURL.Path, storagespace.FormatResourceID(*root))
+		relativePath := res.GetInfo().GetPath()
+		webDavURL.Path = path.Join(webDavURL.Path, storagespace.FormatResourceID(*res.GetInfo().GetSpace().GetRoot()), relativePath)
```

Comment:
This could panic when `...GetSpace()` is not nil but `...GetSpace().GetRoot()` is